otherlibs/stdune/src/io.ml

```diff
@@ -254,32 +254,92 @@ struct
        it fails for w/e reason *)
     external copyfile : string -> string -> unit = "stdune_copyfile"
 
-    external available : unit -> bool = "stdune_is_darwin"
+    external sendfile :
+      src:Unix.file_descr -> dst:Unix.file_descr -> int -> unit
+      = "stdune_sendfile"
+
+    external darwin : unit -> bool = "stdune_is_darwin"
+
+    external linux : unit -> bool = "stdune_is_linux"
+
+    let available =
+      if darwin () then `Copyfile else if linux () then `Sendfile else `Nothing
+
+    let sendfile =
+      let setup_copy ?(chmod = Fun.id) ~src ~dst () =
+        match Unix.openfile (Path.to_string src) [ O_RDONLY ] 0 with
+        | exception Unix.Unix_error (Unix.ENOENT, _, _) -> Error `Src_missing
+        | ic ->
+          let open Result.O in
+          let+ size, oc =
```

Comment:
Why do we care about the `size` of `dst`, and isn't it supposed to be 0 since we create it?

Reply:
It's the size of `src` actually. We need it because `sendfile` needs to know how many bytes to write.

Reply:
Oh, sorry, the naming got me confused. Of course, we need to know the `src`'s size.

Reply:
(Maybe call it `src_size`.)